src/vm/core.zig

```diff
@@ -256,6 +256,12 @@ pub const CairoVM = struct {
         return self.runInstruction(allocator, &instruction);
     }
 
+    pub fn insertDeducedOperands(self: *Self, allocator: Allocator, op: OperandsResult) !void {
```

Comment:
Can we add unit tests to confirm the logic here?

Reply:
Let me know if we need more tests